sdk/search/search-documents/test/public/node/searchClient.spec.ts

```diff
@@ -77,6 +83,122 @@ versionsToTest(serviceVersions, {}, (serviceVersion, onVersions) => {
       assert.equal(searchResults.count, 6);
     });
 
+    it("search narrows the result type", async function () {
+      // eslint-disable-next-line no-constant-condition
+      if (false) {
```

Comment:
is this for debugging or something? can it be re-written in a more actionable way?

Reply:
This is actually here to validate compile-time type behavior. I'm not sure if there's a way to test this using mocha as opposed to letting the compiler fail. Maybe forking a `tsc` instance and checking the exit code? I can look into ways to improve this.